src/dotenv.cr

```diff
@@ -111,11 +21,54 @@ module Dotenv
   # hash # => {"VAR" => "Hello"}
   # ```
   def parse(env_vars : String | IO) : Hash(String, String)
-    hash = Hash(String, String).new
-    env_vars.each_line do |line|
-      handle_line line, hash
+    Parser.new(env_vars).parse
+  end
+
+  # Builds a `.env` formatted string, and escape special characters in values.
+  #
+  # Only variable key characters are validated.
+  #
+  # ```
+  # require "dotenv"
+  #
+  # Dotenv.build({"VAR" => "Hello"}) => "VAR=Hello"
+  # ```
+  def build(env_vars : Hash(String, String), value_quotes : Quotes = Quotes::None) : String
+    String.build { |str| build str, env_vars, value_quotes }
+  end
+
+  # Builds a `.env` formatted data to the `IO`, and quotes to put around the value.
+  #
+  # Only variable key characters are validated.
+  #
+  # ```
+  # require "dotenv"
+  #
+  # File.open "w", ".env" do |io|
+  #   Dotenv.build(io, {"VAR" => "Hello"})
+  # end
+  # ```
+  def build(io, env_vars : Hash(String, String), value_quotes : Quotes = Quotes::None) : Nil
```

Comment:
Since we have a `Parser` wouldn't it makes sense to have a `Builder` as well?

Reply:
Yes, maybe, but I didn't see a need.
For now, the builder is fairly simple.